monopoly/src/main/java/gameset/functionality/Player.java

```diff
@@ -103,11 +103,21 @@ public void addProperty(Property property, Board board) throws IOException {
      * it will then check the value to the color. If it is true, then that means
      * that they have monopoly.
      *
-     * @param color the property that would be added
+     * @param property The property that the building will be placed on.
+     * @param board The board object
      */
-    public boolean canPurchaseBuilding(String color){
-        if(colorSets.containsKey(color)){
-            return colorSets.get(color);
+    public boolean canPurchaseBuilding(Property property, Board board) throws IOException {
+        if(colorSets.containsKey(property.getColor())){
+            if (colorSets.get(property.getColor())) {
```

Comment:
Can these 2 conditionals be in same statement? It might be easier to read, I thought there was something else happening with the second conditional (if colorset.get was false), but that doesn't seem to be the case.
```suggestion
        if(colorSets.containsKey(property.getColor()) && colorSets.get(property.getColor())){
```